src/groundStation/commandParser.py

```diff
@@ -118,8 +118,11 @@ def parseReturnValue(self, src, dst, dport, data, length):
                 return outputObj
 
             else:
-                outputObj[retVal] = np.frombuffer(
-                    data, dtype=returns[retVal], count=1, offset=idx)[0]
+                try:
+                    outputObj[retVal] = np.frombuffer(
+                        data, dtype=returns[retVal], count=1, offset=idx)[0]
+                except:
+                    return outputObj
```

Comment:
In this error case can outputObj be queried to find out what went wrong?

Reply:
I don't think I understand python well enough to answer that

Reply:
It should return outputObj when numpy can't extract the return value but (I don't usually see this side of functionality so I could be wrong) it might be hard to tell when something goes wrong as the error message is suppressed when using a try/except (if an error occurs, it immediately goes to the except block without outputting the error).

Reply:
The error happens when the buffer is not the expected size, so numpy freaks out about it. With our use we usually send a shorter buffer if there's an error because there's no data to fill, rather than using dummy bytes. I'm not sure how what you're saying fits with this, probably because of my lack of python experience

Reply:
Oh I was just saying that if you don't know what you're looking for, it might be hard (haven't seen this side of operation so I could be wrong) to distinguish an error from normal behaviour since they'll both return outputObj, though I guess outputObj[retVal] might be empty.

Reply:
Oh I see. The idea would be to check the 'err' value first before trying to access other things. Though I have no idea if this is good programming

Reply:
Actually, can you think of a way I could fill in None for each outputObj[retval] that isn't there?